### PR TITLE
The New Quick Tour

### LINKED ISSUE #8884: Update the Quick Tour for Symfony 4
Part of the "upgrade to Symfony 4" initiative (#8754) which will be treated separately because it will require some time.

### PR DESCRIPTION
**Pull request by weaverryan**
Fixes #8884 

Hi guys!

The quick tour will now be 3 parts (instead of 4), and I've finished the first 2 parts. I'm purposely using the first person point of view and more conversational language.

Cheers!

### CODE REVIEW

File: quick_tour/the_big_picture.rst [hunk 1]
```diff
@@ -1,267 +1,182 @@
...
-Start using Symfony in 10 minutes! This chapter will walk you through the
-most important concepts behind Symfony and explain how you can get started
-quickly by showing you a simple project in action.
+Start using Symfony in 10 minutes! Really! That's all you need to understand the
+most important concepts and start building a real project!
 
 If you've used a web framework before, you should feel right at home with
-Symfony. If not, welcome to a whole new way of developing web applications.
+Symfony. If not, welcome to a whole new way of developing web applications. Symfony
+*embraces* best practices, keeps backwards compatbility (Yes! Upgrading is always
```
Comment (javiereguiluz) [inline quick_tour/the_big_picture.rst]: typo: `compatbility`

File: quick_tour/the_big_picture.rst [hunk 2]
```diff
@@ -1,267 +1,182 @@
...
-to the *resource* associated with it (the ``Homepage`` HTML page).
+    $ composer create-project symfony/skeleton quick_tour
 
-The code to execute is defined as methods of PHP classes. The methods are
-called **actions** and the classes **controllers**, but in practice most
-developers use **controllers** to refer to both of them. The mapping between
-user's requests and that code is defined via the **routing** configuration.
-And the contents displayed in the browser are usually rendered using
-**templates**.
+This creates a new ``quick_tour/`` directory with, at first, just *one* file:
```
Comment (javiereguiluz) [inline quick_tour/the_big_picture.rst]: I find this explanation about having just 1 file called `composer.json` confusing. Users won't ever see this single file but the entire project files, so I'd recommend to remove this.

File: quick_tour/the_big_picture.rst [hunk 3]
```diff
@@ -1,267 +1,182 @@
...
+    ├─ config/
+    ├─ public/index.php
+    ├─ src/
+    ├─ symfony.lock
+    ├─ var/
+    └─ vendor/
 
-Actions and Controllers
-~~~~~~~~~~~~~~~~~~~~~~~
+Where did these files come from? They came from recipes installed by Symfony Flex.
```
Comment (javiereguiluz) [inline quick_tour/the_big_picture.rst]: I thin kit's too early to mention concepts like Flex and recipes. They run a "create-project" command and they got a project created. That's intuitive. If we add this mention to Flex/recipes, things complicate without a clear benefit.

File: quick_tour/the_big_picture.rst [hunk 4]
```diff
@@ -1,267 +1,182 @@
...
+But don't worry about that yet: we'll talk about it soon.
 
-Open the ``src/Controller/DefaultController.php`` file and you'll
-see the following code (for now, don't look at the ``@Route`` configuration
-because that will be explained in the next section)::
+Can we already load the project in a browser? Of course! You can setup
+:doc:`Nginx or Apache <setup/web_server_configuration>` and configure their document
+root to be the ``public/`` directory. But, for development, Symfony has its own server.
+Install it with:
```
Comment (javiereguiluz) [inline quick_tour/the_big_picture.rst]: `Install it and run it with:` ?

File: quick_tour/flex_recipes.rst [hunk 5]
```diff
@@ -0,0 +1,249 @@
+Flex: Compose your Application
+==============================
+
+After reading the first part of this tutorial, you have decided that Symfony was
+worth another 10 minutes. Great choice! In this second part, you'll learn about
+Symfony Flex: the amazing tool that makes adding new features as simple as running
+one command. It's also the reason why Symfony is perfect for a small micro-service
```
Comment (javiereguiluz) [inline quick_tour/flex_recipes.rst]: Just asking: can this sound arrogant? --> *"[...] Symfony is perfect for [...]"* Could we reword it?

File: quick_tour/flex_recipes.rst [hunk 6]
```diff
@@ -0,0 +1,249 @@
...
+
+First, ``twig`` is not the name of a Composer package: it's a Flex *alias* that
+points to ``symfony/twig-bundle``. Flex resolves that alias for Composer.
+
+And second, Flex installs a *recipe* for ``symfony/twig-bundle``. What's a recipe?
+It's a way for a library to automatically configure itself by adding and modifying
+files. Thanks to recipes, adding features is seamless and automated: install a package
+and you're done!
+
+You an find a full list of recipes and aliases by going to `https://symfony.sh`_.
```
Comment (javiereguiluz) [inline quick_tour/flex_recipes.rst]: typo: `an find` -> `can find`

File: quick_tour/flex_recipes.rst [hunk 7]
```diff
@@ -0,0 +1,249 @@
...
+
+What did this recipe do? In addition to automatically enabling the bundle in
```
Comment (javiereguiluz) [inline quick_tour/flex_recipes.rst]: *"enabling the bundle"* <-- will people understand that? It's the first time that "bundle" is mentioned in the Quick Tour. Maybe: *"enabling the feature"* is better?

File: quick_tour/flex_recipes.rst [hunk 8]
```diff
@@ -0,0 +1,249 @@
...
+``config/bundles.php``, it added 3 things:
+
+``config/packages/twig.yaml``
+    A configuration file that sets up Twig with sensible defaults.
+
+``config/routes/dev/twig.yaml``
+    A route that helps you debug your error pages: it's added only in the ``dev``
```
Comment (javiereguiluz) [inline quick_tour/flex_recipes.rst]: This is the first time that *"environment"* is mentioned. Maybe we can remove this entirely -> *"it's added only in the ``dev`` environment."*

File: quick_tour/flex_recipes.rst [hunk 9]
```diff
@@ -0,0 +1,249 @@
...
+----------------------------
+
+One of the *coolest* features of Symfony isn't even installed yet! Let's fix that:
+
+.. code-block:: terminal
+
+    $ composer require profiler
+
+Yes! This is another alias! And Flex *also* installs another recipe, which automates
+the configuration of Symfony's WebProfilerBundle. What's the result? Refresh!
```
Comment (javiereguiluz) [inline quick_tour/flex_recipes.rst]: I'd prefer to never mention bundles in the Quick tour. What if we replace `Symfony's WebProfilerBundle` by `Symfony's Profiler` ?

File: quick_tour/flex_recipes.rst [hunk 10]
```diff
@@ -0,0 +1,249 @@
...
+Not convinced yet? No problem: remove the library:
+
+.. code-block:: terminal
+
+    $ composer remove api
+
+Flex will *uninstall* the recipes: removing files and un-doing changes to put your
+app back in its original state. Experiment without worry.
+
+More Feature, Architecture and Speed
```
Comment (javiereguiluz) [inline quick_tour/flex_recipes.rst]: typo? `More Feature, ...` -> `More Features, ...`

**Review by javiereguiluz — COMMENTED** [review]
This is really nice. Chapters are short, engaging and they build hype gradually. Fantastic work so far. I can't wait to read the rest of chapters. Thanks!

**Comment by weaverryan** [comment]
Thanks for the fast review Javier! And now it's fully finished and the last chapter is ready for review :)

File: quick_tour/the_architecture.rst [hunk 11]
```diff
@@ -1,4 +1,139 @@
...
+    // ...
+
+    public function index($name, LoggerInterface $logger)
+    {
+        $logger->info("Saying hello to $name!");
+
+        // ...
+    }
+
+That's it! The new log message be written to ``var/log/dev.log``. Of course, this
```
Comment (javiereguiluz) [inline quick_tour/the_architecture.rst]: `message be written` -> `message will be written` ?

File: quick_tour/the_architecture.rst [hunk 12]
```diff
@@ -1,4 +1,139 @@
...
+Services & Autowiring
+---------------------
+
+But wait! Something *very* cool just happened. Symfony read the ``LoggerInterface``
+type-hint and automatically figured out that it should pass us the Logger object!
+This is called *autowiring*.
+
+Every bit of work that's done in a Symfony app is done by an *object*: the Logger
+object logs things and the Twig object renders templates. These objects are called
+*services* and they are the *tools* that make you dangerous.
```
Comment (javiereguiluz) [inline quick_tour/the_architecture.rst]: I'm not sure this *"dangerous"* word is going to be understood well by people with basic English level, which are most of our readers. Why if we say something easier to understand, like: *"...the *tools* that give you super powers!"*

File: quick_tour/the_architecture.rst [hunk 13]
```diff
@@ -1,4 +1,139 @@
...
+        {
+            // ...
+
+     +        $this->logger->info('Using the greeting: '.$greeting);
+
+             return $greeting;
+        }
+    }
+
+Yes! This works too: no configuration, no time wasted.
```
Comment (javiereguiluz) [inline quick_tour/the_architecture.rst]: `no configuration` -> `no service configuration needed` or something like that?

**Review by javiereguiluz — APPROVED** [review]
It's great to see this finished!

(and, again, lines deleted are twice the lines added)

**Comment by weaverryan** [comment]
Changes made!

**Pull request merged**